pandas/core/groupby/ops.py

```diff
@@ -491,8 +491,11 @@ def _ea_wrap_cython_operation(
             res_values, names = self._cython_operation(
                 kind, values, how, axis, min_count, **kwargs
             )
+            if how in ["rank"]:
```

Comment:
hmm, this is prob not the best location for this longer term (eg.. we should have mappings from ops -> whether they should coerce back or not).

Reply:
yah, we have something like that in the calling class.  im in the process of moving all of this wrapping to the lowest level possible (which luckily coincides with it being in the fewest places possible), xref https://github.com/pandas-dev/pandas/issues/37494#issuecomment-736813064

Reply:
Could this also be handled in `maybe_cast_result_dtype`? (which basically is somewhat implementing a mapping from ops -> expected dtype)

Reply:
potentially, yah.  this calls simple_new instead of from_sequence, which makes it a bit of an outlier.

there's also groupby.base.cython_cast_blocklist that i think is intended for something along these lines